src/data/fin.lean

```diff
@@ -1177,6 +1194,22 @@ begin
       exact lt_succ_iff.mpr (not_lt.mp h), }, },
 end
 
+/-- `pred` commutes with `succ_above`. -/
+lemma pred_succ_above_pred {a : fin (n + 2)} {b : fin (n + 1)} (ha : a ≠ 0) (hb : b ≠ 0) (hk) :
```

Comment:
Since `hk` follows from the other assumptions, we can give it a default value so the user doesn't need to supply it in practice:
```suggestion
lemma pred_succ_above_pred {a : fin (n + 2)} {b : fin (n + 1)} (ha : a ≠ 0) (hb : b ≠ 0)
  (hk := succ_above_ne_zero ha hb) :
```
(see the other suggestion for a (hopefully correct) proof of `succ_above_ne_zero`)).

Reply:
Ooh, I didn't know default values were a thing; that's awesome!